### PR TITLE
fix(autocomplete): up arrow should set last item active

### PR DESCRIPTION
**Pull request by kara**
r: @jelbourn @andrewseguin

### CODE REVIEW

File: src/lib/autocomplete/autocomplete.spec.ts [hunk 1]
```diff
@@ -553,5 +676,15 @@ class FakeKeyboardEvent {
   preventDefault() {}
 }
 
+class FakeViewportRuler {
```
Comment (jelbourn) [inline src/lib/autocomplete/autocomplete.spec.ts]: Isn't there a common `FakeviewportRuler` now?

File: src/lib/autocomplete/autocomplete-trigger.ts [hunk 2]
```diff
@@ -5,18 +5,28 @@ import {NgControl} from '@angular/forms';
...
+ * actually focusing the active item, scroll must be handled manually.
+ */
 
-/** The panel needs a slight y-offset to ensure the input underline displays. */
-export const MD_AUTOCOMPLETE_PANEL_OFFSET = 6;
+/** The height of each autocomplete option. */
+export const AUTOCOMPLETE_OPTION_HEIGHT = 48;
+
+/** The total height of the autocomplete panel. */
+export const MD_AUTOCOMPLETE_PANEL_HEIGHT = 256;
```
Comment (kara) [inline src/lib/autocomplete/autocomplete-trigger.ts]: Good catch; not sure why I did that.

File: src/lib/autocomplete/autocomplete.spec.ts (hunk 1 above)
Comment (kara) [inline src/lib/autocomplete/autocomplete.spec.ts]: Oh hey. Didn't know about that, but that's way better. Will replace.

File: src/lib/core/a11y/list-key-manager.ts [hunk 3]
```diff
@@ -96,12 +96,13 @@ export class ListKeyManager<T extends CanDisable> {
...
   setNextItemActive(): void {
-    this._setActiveItemByDelta(1);
+    this._activeItemIndex === null ? this.setFirstItemActive() : this._setActiveItemByDelta(1);
   }
 
   /** Sets the active item to a previous enabled item in the list. */
   setPreviousItemActive(): void {
-    this._setActiveItemByDelta(-1);
+    this._activeItemIndex === null && this._wrap ? this.setLastItemActive()
+                                                 : this._setActiveItemByDelta(-1);
```
Comment (jelbourn) [inline src/lib/core/a11y/list-key-manager.ts]: Need to handle the case when `activeItemIndex` is null and wrap is false

Comment (jelbourn) [inline src/lib/core/a11y/list-key-manager.ts]: (with a unit test)

**Comment by jelbourn** [comment]
LGTM

**Pull request merged**